src/webring.rs

```diff
@@ -387,77 +406,123 @@ impl Webring {
-                                    info!("Webring reloaded");
-                                });
+                        // Keep track of which kind of reload to perform, so we only do one per
+                        // group of paths changed.
+                        let mut reload_kind = None;
+
+                        for path in &event.paths {
+                            // Handle homepage template update
+                            if files_match(path, &homepage_template_for_closure) {
+                                if reload_kind.is_none() {
+                                    reload_kind = Some(LiveReloadEvent::Homepage);
+                                }
                             }
-                            if same_file::is_same_file(&path, &homepage_template_path_for_closure)
-                                .unwrap_or_else(|err| {
-                                    log::error!("Error comparing file paths: {err}");
-                                    false
-                                })
-                            {
-                                info!(
-                                    "Detected change to {}. Invalidating homepage.",
-                                    homepage_template_path_for_closure.display()
-                                );
-                                let webring_for_task = Arc::clone(&webring);
-                                rt.spawn(async move {
-                                    webring_for_task.homepage.write().await.take();
-                                });
+                            // Handle config file update
+                            else if files_match(path, &config_file_for_closure) {
+                                reload_kind = Some(LiveReloadEvent::Config);
+                            }
+                            // A file we weren't watching was updated?
+                            else {
+                                log::warn!("Unexpected file updated: {}", path.display());
                             }
                         }
+
+                        if let Some(kind) = reload_kind {
+                            let config_file_path_for_task = config_file_for_closure.clone();
+                            // This must succeed, because if the webring was dropped, so would be
+                            // the thread running the event handler.
+                            let webring = weak_webring.upgrade().unwrap();
+                            rt.spawn(async move {
+                                // This must succeed, because the webring
+                                match kind {
+                                    LiveReloadEvent::Config => {
+                                        log::info!("Configuration file updated; reloading members");
+                                        // Load new config
+                                        match Config::parse_from_file(&config_file_path_for_task)
```

Comment:
I'm not sure how much this matters, but I would be worried about someone thinking that automatic reloading applies to all of the settings in the config file and getting confused when it doesn't update properly. Would it be possible/easy to check if any other options were updated and if so log a warning? If not it's probably nbd.

Reply:
Agreed. I meant to mention that in the documentation. Do you think that would be enough? Diffing the configs sounds a little annoying to implement.

Reply:
You could store the config in the webring structure and have a method on the config struct that checks for equality on all of the fields except for the members field. Then you could use that method to tell whether anything else has changed.

Reply:
Yeah that's what I had in mind. I was just too lazy to implement it earlier, but I've added it now.